stdlib/zipfile.pyi

```diff
@@ -274,6 +274,12 @@ if sys.version_info >= (3, 8):
             def joinpath(self, *other: StrPath) -> Path: ...
         else:
             def joinpath(self, add: StrPath) -> Path: ...  # undocumented
+        if sys.version_info >= (3, 12):
+            def glob(self, pattern: str) -> map[Self]: ...
+            def rglob(self, pattern: str) -> map[Self]: ...
```

Comment:
These look to me like they return generator expressions, not instances of `map`: https://github.com/python/cpython/commit/36854bbb240e417c0df6f0014924fcc899388186#diff-06a1fd516348dcbb784169d31d76c5171caedf093fb423ce2d08a4442452d013R359-R371

```suggestion
            def glob(self, pattern: str) -> Iterator[Self]: ...
            def rglob(self, pattern: str) -> Iterator[Self]: ...
```

Also, it looks like our annotation for `iterdir()` in this file is wrong: it should also return `Iterator[Self]`, but currently returns `Iterator[Path]`. Would you like to also correct that in this PR?

Reply:
This was the source of my `map` proposal. I still think that current runtime behaves incorrectly, but I don't like to persuade people, it literally drains life out of me :)

Reply:
> but I don't like to persuade people, it literally drains life out of me :)

I can empathise with that. I'm sorry that I disagreed with your proposal this time :)

Reply:
No worries, I find motivation in my own projects :)

Reply:
One of the reasons I created typeshed-stats was so I could have a project nobody could argue with me about 😄